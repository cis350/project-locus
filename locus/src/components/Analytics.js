/* eslint-disable no-underscore-dangle */
import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import {
  Button,
  Card,
} from 'react-bootstrap';
import { getAllTasks } from '../modules/api';

const Analytics = function AnalyticsComponent({
  userId,
  currProjectWithoutSpace,
  project,
  club,
  email,
}) {
  const [allTasks, setAllTasks] = useState([]);
  useEffect(() => {
    getAllTasks(project, club, email).then((res) => {
      if (res.status === 200) {
        setAllTasks(res.jsonContent);
      }
    });
  }, []);

  const navigate = useNavigate();

  const returnToProject = (() => {
    navigate(`/projects/manage-projects/${currProjectWithoutSpace}/${userId}`);
  });

  const tasksCompleted = (() => (
    allTasks.map((data) => {
      if (data.status === 'done') {
        console.log(data);
        return (
          <li style={{ listStyleType: 'none', width: '50rem' }} key={data._id}>
            <div className="row">
              <div className="col-6">
                {data.taskName}
              </div>
            </div>
          </li>
        );
      }
    })
  ));

  const tasksIncompleted = (() => (
    allTasks.map((data) => {
      if (data.status === 'incomplete') {
        console.log(data);
        return (
          <li style={{ listStyleType: 'none', width: '50rem' }} key={data._id}>
            <div className="row">
              <div className="col-6">
                {data.taskName}
              </div>
            </div>
          </li>
        );
      }
      if (data.status === 'need help') {
        console.log(data);
        return (
          <li style={{ listStyleType: 'none', width: '50rem' }} key={data._id}>
            <div className="row">
              <div className="col-6">
                {data.taskName}
                {' '}
                (need help)
              </div>
            </div>
          </li>
        );
      }
    })
  ));

  return (
    <div className="container" style={{ position: 'relative', padding: '20px' }}>
      <div className="row">
        <div className="col-sm">
          <Card style={{
            width: '23rem',
            height: '27rem',
            margin: 'auto',
            marginTop: '20px',
            borderRadius: '10px',
            backgroundColor: '#B5E48C',
            borderColor: '#B5E48C',
          }}
          >
            <Card.Body>
              {/* referenced https://react-bootstrap.github.io/forms/overview/ */}
              <h5 className="text-center">Completed Tasks</h5>
              <div style={{ overflow: 'hidden', overflowY: 'scroll' }}>
                <ul style={{ height: '310px' }}>
                  {tasksCompleted()}
                </ul>
              </div>
            </Card.Body>
          </Card>
        </div>
        <div className="col-sm">
          <Card style={{
            width: '23rem',
            height: '27rem',
            margin: 'auto',
            marginTop: '20px',
            borderRadius: '10px',
            backgroundColor: '#B5E48C',
            borderColor: '#B5E48C',
          }}
          >
            <Card.Body>
              {/* referenced https://react-bootstrap.github.io/forms/overview/ */}
              <h5 className="text-center">Incompleted Tasks</h5>
              <div style={{ overflow: 'hidden', overflowY: 'scroll' }}>
                <ul style={{ height: '310px' }}>
                  {tasksIncompleted()}
                </ul>
              </div>
            </Card.Body>
          </Card>
        </div>
      </div>
      <div className="navbar-brand" style={{ display: 'flex', justifyContent: 'center', marginTop: '30px' }}>
        <Button
          style={{
            backgroundColor: '#C30000',
            width: '120px',
            height: '50px',
            fontWeight: 'bold',
            fontSize: '25px',
            color: 'black',
            borderColor: '#C30000',
          }}
          onClick={() => returnToProject()}
        >
          Return
        </Button>
      </div>
    </div>
  );
};

export default Analytics;
