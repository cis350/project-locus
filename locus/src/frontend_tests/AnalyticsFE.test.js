/**
* @jest-environment jsdom
*/
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Analytics from '../components/Analytics';

const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');

// mock useNavigate: https://github.com/remix-run/react-router/issues/7811
const mockedNavigator = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedNavigator,
}));

let mock;
beforeAll(() => {
  mock = new MockAdapter(axios);
});
afterEach(() => {
  mock.reset();
});

test('analytics snapshot test', () => {
  const component = renderer.create(<Analytics />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
