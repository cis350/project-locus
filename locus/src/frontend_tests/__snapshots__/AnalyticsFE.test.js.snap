// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analytics snapshot test 1`] = `
<div
  className="container"
  style={
    Object {
      "padding": "20px",
      "position": "relative",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <div
        className="card"
        style={
          Object {
            "backgroundColor": "#B5E48C",
            "borderColor": "#B5E48C",
            "borderRadius": "10px",
            "height": "27rem",
            "margin": "auto",
            "marginTop": "20px",
            "width": "23rem",
          }
        }
      >
        <div
          className="card-body"
        >
          <h5
            className="text-center"
          >
            Completed Tasks
          </h5>
          <div
            style={
              Object {
                "overflow": "hidden",
                "overflowY": "scroll",
              }
            }
          >
            <ul
              style={
                Object {
                  "height": "310px",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm"
    >
      <div
        className="card"
        style={
          Object {
            "backgroundColor": "#B5E48C",
            "borderColor": "#B5E48C",
            "borderRadius": "10px",
            "height": "27rem",
            "margin": "auto",
            "marginTop": "20px",
            "width": "23rem",
          }
        }
      >
        <div
          className="card-body"
        >
          <h5
            className="text-center"
          >
            Incompleted Tasks
          </h5>
          <div
            style={
              Object {
                "overflow": "hidden",
                "overflowY": "scroll",
              }
            }
          >
            <ul
              style={
                Object {
                  "height": "310px",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="navbar-brand"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "marginTop": "30px",
      }
    }
  >
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#C30000",
          "borderColor": "#C30000",
          "color": "black",
          "fontSize": "25px",
          "fontWeight": "bold",
          "height": "50px",
          "width": "120px",
        }
      }
      type="button"
    >
      Return
    </button>
  </div>
</div>
`;
